README.md

```diff
@@ -86,7 +86,150 @@ There are several ways to execute the code cells in your notebook:
+        <div><strong>Layers</strong></div>
+        {% for layer in this.layers %}
+        <div class="rendererOpt checkbox checkbox-primary">
+            <input type="checkbox" pd_refresh="map{{prefix}}" pd_script="self.toggleLayer({{loop.index0}})">
+            <label>{{layer["name"]}}</label>
+        </div>      
+        {%endfor%}
+    </div>
+    <div class="form-group col-sm-10">
+        <div id="map{{prefix}}" pd_entity pd_options="{{this.formatOptions(this.mapJSONOptions)}}"/>
+    </div>
+</div>
+"""
+```
+#### Create the Map of Accidents
+```
+<div id="map{{prefix}}" pd_entity pd_options="{{this.formatOptions(this.mapJSONOptions)}}"/>
+```
+pd_entity: tell PixieDust which dataset to work on 
+ 
+pd_options: Contains the PixieDust options for the map 
+#### Generate the pd_options
+
+The best way to generate the pd_options for a PixieDust visualization is to: 
+1.  Call display() on a new cell 
+2.  Graphically select the options for your chart 
+3.  Select View/Cell Toobar/Edit metadata menu 
+4.  Click on the “Edit Metadata” button and copy the pixiedust metadata  
+
+![](docs/sources/pixieEditMetadata.png)
+
+ To conform to the pd_options notation, we need to transform the PixieDust JSON metadata into an attribute string with the following format: ```“key1=value1;key2=value2;…”```
+
+
+To make it easier, we use the a simple python transform function: 
+```
+def formatOptions(self, options):
+    return ';'.join(["{}={}".format(k,v) for (k, v) in iteritems(options)])
+```
+
+The formatOptions is then invoked using JinJa2 notation from within the html:
+```
+pd_options = “{{this.formatOptions(this.mapJSONOptions)}}” 
+```
+#### Initialize the pd_options
+
+ Note: setup is a special method that will be called automatically when the PixieApp is initialized.
+
+```
+   def setup(self):
+        self.mapJSONOptions = {
+          "mapboxtoken": "pk.eyJ1IjoicmFqcnNpbmdoIiwiYSI6ImNqM2s4ZDg4djAwcGYyd3BwaGxwaDV3bWoifQ.d5Rklkdu5MeGAnXu1GMNYw",
+          "chartsize": "90",
+          "aggregation": "SUM",
+          "rowCount": "500",
+          "handlerId": "mapView",
+          "rendererId": "mapbox",
+          "valueFields": "IncidntNum",
+          "keyFields": "X,Y",
+          "basemap": "light-v9"
+        }
+```
+
+#### Create the GeoJSON Custom Layers
+```
+from pixiedust.display.app import *
+from pixiedust.apps.mapboxBase import MapboxBase
+
+@PixieApp
+class SFDashboard(MapboxBase):
+    def setup(self):
+..<snip>..  
+self.setLayers([
+        {
+            "name": "Speeding",
+            "url": "https://data.sfgov.org/api/geospatial/mfjz-pnye?method=export&format=GeoJSON"
+        },
+        {
+            "name": "Traffic calming",
+            "url": "https://data.sfgov.org/api/geospatial/ddye-rism?method=export&format=GeoJSON",
+            "type": "symbol",
+            "layout": {
+                "icon-image": "police-15",
+                "icon-size": 1.5
+            }
+        },
+..<snip>..
```

Comment:
two .. bothers me.  Use ... or choose to bug me.  I think it is a math thing (or just a peeve).